src/utilities/timeout.inc

```diff
@@ -0,0 +1,120 @@
+#!/bin/bash
+set -e
+
+# 5 minutes by default
+default_timeout=300
+export TIMEOUT=$default_timeout
+
+# 
+create_test_file() {
+
+    if [ "$#" -eq 1 ]
+    then
+	    rm -f "$1"
+	    echo "#!/bin/bash" >> "$1"
+        return 0
+    else
+        echo "Error. Missing the parameter for the test file to create."
+        return 1
+    fi
+}
+
+# Set the timeout.
+# $1: timeout in seconds. A negative value is not allowed. Set 0 to disable the timeout and -1 to reset to default value
+set_timeout() {
+
+    if [ "$#" -eq 1 ]
+    then
+        if [ "$1" -gt 0 ]
+        then
+            export TIMEOUT=$1
+            return 0
+        elif [ "$1" -eq 0 ]
+        then
+            unset TIMEOUT
+            return 0
+        elif [ "$1" -eq -1 ]
+        then
+            export TIMEOUT=$default_timeout
+            return 0
+        else
+            echo "[ERROR] Wrong timeout value. It must be a positive integer, 0 to disable or -1 to reset to default value"
+            return 1
+        fi
+    else
+        echo "Error. Wrong number of parameters. Use timeout timeout_value"
+        echo "   The timeout value is a positive integer. Use '0' to disable the timeout and '-1' to reset to default value"
```

Comment:
It's a little weird saying that it must be positive and then explain it can be -1.